message_ix_models/tools/costs/projections.py

```diff
@@ -397,11 +397,23 @@ def create_message_outputs(
         (
             df_prod.merge(
                 val_2020,
-                on=["scenario_version", "scenario", "message_technology", "region"],
+                on=[
+                    "scenario_version",
+                    "scenario",
+                    "message_technology",
+                    "first_technology_year",
+                    "region",
+                ],
             )
             .merge(
                 val_2100,
-                on=["scenario_version", "scenario", "message_technology", "region"],
+                on=[
```

Comment:
For these chained pandas calls, it is much more readable to define a list of dimensions/column names before the chain:
```py
dims = ["scenario_version", "scenario", ..., "region"]
```
and then reference it:
```py
.merge(val_2020, on=dims)
.merge(val_2100, on=dims)
```

Reply:
Thanks! Have updated as such:

https://github.com/iiasa/message-ix-models/blob/08ccfd6504c8be03e63718889eaa102677d7d6b3/message_ix_models/tools/costs/projections.py#L396-L413